### ISSUE TITLE
Improvement on Theory Part 1 and Code Part 1

### ISSUE THREAD

**Issue by HuwWDay** [Bristol-Training/intro_to_AI#4]
Please *don't* start work on this until you've met with Huw!

Make copies of the notebooks and fix them as outlined below, call the files something different e.g. "2-HiddenLayers.ipynb" should become "2-HiddenLayersFINAL.ipynb"

In TheoryPart1_Boyi.ipynb:
- [ ] right before the text that says "Decision Broundary" (fix that typo aha), introduce a brief note explaining how in practice, we don't usually output a single number, instead our output layer is usually just some numbers that correspond to (for example) the number of classes in our classification problem and re-scaled (using softmax) so that we can see something for the analogy probability of our neural network placing each data point in each class. This is particularly relevant when we classify to more than 2 classes.

In 2-HiddenLayers_Boyi.ipynb:
- [ ]  right at the top below the first bit of text, add a diagram of a multilayer perceptron with two inputs x1 and x2 with one hidden layer with two neurons and one output neuron.
- [ ] the decision boundary code for "decision_boundary_toy_mlp" is happily working, but I think my function "toy_MLP_with_activations" near the bottom isn't working properly and so is unable to integrate with it. Can you please fix this so that students are able to visualise decision boundaries using different activation functions, weights and biases?
- [ ] When we mention decision boundary, literally at the bottom of the page add a note that say in a feature space of 2 dimensions, the decision boundary is a line, but in higher dimensions, the decision boundary can be a plane (3D) or some sort of hyperplane (4D) that separates points. There's also no reason 

 In intro_to_AI_zhengzhe.ipynb:
- [ ] I'd like to make the intro to tensors + Pytorch shorter. But I think it's better to only show a few key things and then link to further documentation. E.g. Make zero, one and random tensor, return the shape, add tensors, matrix mult (with a 2x2 example) and dot products. This is not because you've written anything wrong, just in the interest of time!
- [ ] Make a new notebook that starts with the introduction of classes and perceptron. I don't mind what you call it, but make sure it's clear what sequence the notebooks go into.
- [ ] Make a new notebook starting from DataPipeline (I basically want to be able to sort the course into clear "chapters") 
- [ ] When you load in the iris dataset, can you add one or two plots just showing what it looks like. We want students watching to think "if I got some data and wanted to put it into a neural network, first I should just have a look at the data to see what it does" 
See: https://bristol-training.github.io/applied-data-analysis-in-python/pages/102-nearest-neighbours.html and https://bristol-training.github.io/introduction-to-data-analysis-in-python/pages/05%20Visualisation.html for some examples (maybe a pair plot, and df.head())
- [ ] Make a new notebook starting from Multi-layer Perceptron Implementation
- [ ] Consider if the intro text about the FNN can be shortened a little bit
- [ ] Remove step 5 (we don't do gradient descent until theory and code part 2!)
